📖Add page in book for how to watch objects referenced in the CRD. (#2000)
kubernetes-sigs/kubebuilder: docs/book/src/reference/watching-resources.md

# Watching Resources

> Inside a `Reconcile()` control loop, controllers will often use the state of the kubernetes cluster, and resources in it, to make decisions.
> Therefore, it can be necessary to know when a resource that you care about is changed.

Reviewer: ```suggestion
Inside a `Reconcile()` control loop, you are looking to do a collection of operations until it has the desired state on the cluster. Therefore, it can be necessary to know when a resource that you care about is changed.
```

> The [Controller Runtime libraries](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/builder) provide many ways for resources to be managed, and watched.
> This ranges from the easy and obvious use cases, such as watching the resources you are controlling,
> to more unique and advanced use cases.

Reviewer: ```suggestion
[Controller Runtime libraries](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/builder) provide many ways for resources to be managed and watched. This ranges from the easy and obvious use cases, such as watching the resources which were created and managed by the controller to more unique and advanced use cases.
```
